lib/blob/blobstore.c

```diff
@@ -700,6 +703,7 @@ blob_parse_page(const struct spdk_blob_md_page *page, struct spdk_blob *blob)
 					if (desc_extent_rle->extents[i].cluster_idx != 0) {
 						blob->active.clusters[blob->active.num_clusters++] = bs_cluster_to_lba(blob->bs,
 								desc_extent_rle->extents[i].cluster_idx + j);
+						blob->active.num_allocated_clusters++;
```

Comment:
Can you elaborate on the update of num_allocated_clusters?

Reply:
`blob->active.num_allocated_clusters` keep track of the number of elements != 0 in the `blob->active.clusters` array.
Here we set an element of the array, so we must increment `blob->active.num_allocated_clusters` as well.
This function is called when a superblock is found in a bdev and so the raid is created from it.

Reply:
Keeping track of `active.num_allocated_clusters` let us to avoid counting the number of elements != 0 in the `active.clusters` array every time the function `spdk_blob_get_num_allocated_clusters` is called.